$NetBSD: patch-SP_Makefile,v 1.3 2024/05/12 17:51:54 nia Exp $

Pass through ARCH_STRING properly on *BSD.
Generic platforms (IRIX with clang): link the client and renderers with SDL and curl.

--- SP/Makefile.orig
+++ SP/Makefile
@@ -8,8 +8,8 @@
 COMPILE_ARCH=$(shell uname -m | sed -e 's/i.86/x86/' | sed -e 's/^arm.*/arm/')
 ARM_VER_CHECK=$(shell uname -m)
 
-ifeq ($(COMPILE_PLATFORM),sunos)
-  # Solaris uname and GNU uname differ
+ifneq (,$(findstring "$(COMPILE_PLATFORM)", "sunos" "netbsd"))
+  # Solaris/NetBSD uname and GNU uname differ
   COMPILE_ARCH=$(shell uname -p | sed -e 's/i.86/x86/')
 endif
 
@@ -358,7 +358,7 @@
 EXTRA_FILES=
 CLIENT_EXTRA_FILES=
 
-ifneq (,$(findstring "$(COMPILE_PLATFORM)", "linux" "gnu_kfreebsd" "kfreebsd-gnu" "gnu"))
+ifneq (,$(findstring "$(COMPILE_PLATFORM)", "linux" "bsd" "gnu"))
   TOOLS_CFLAGS += -DARCH_STRING=\"$(COMPILE_ARCH)\"
 endif
 
@@ -563,7 +563,7 @@
 
   SHLIBEXT=dylib
   SHLIBCFLAGS=-fPIC -fno-common
-  SHLIBLDFLAGS=-dynamiclib $(LDFLAGS) -Wl,-U,_com_altivec
+  SHLIBLDFLAGS=-dynamiclib $(LDFLAGS)
 
   NOTSHLIBCFLAGS=-mdynamic-no-pic
 
@@ -741,7 +741,8 @@
 ifneq (,$(findstring "$(PLATFORM)", "freebsd" "openbsd" "netbsd"))
 
   BASE_CFLAGS = -Wall -fno-strict-aliasing \
-    -pipe -DUSE_ICON -DMAP_ANONYMOUS=MAP_ANON
+    -pipe -DUSE_ICON -DMAP_ANONYMOUS=MAP_ANON \
+    -DARCH_STRING=\\\"$(FILE_ARCH)\\\"
   CLIENT_CFLAGS += $(SDL_CFLAGS)
 
   OPTIMIZEVM = -O3
@@ -750,7 +751,6 @@
   ifeq ($(ARCH),x86_64)
     OPTIMIZEVM = -O3
     OPTIMIZE = $(OPTIMIZEVM) -ffast-math
-    FILE_ARCH = amd64
   endif
   ifeq ($(ARCH),x86)
     OPTIMIZEVM = -O3 -march=i586
@@ -892,6 +892,16 @@
   SHLIBCFLAGS=-fPIC
   SHLIBLDFLAGS=-shared
 
+  # The client and renderers need SDL (and curl) here as on every other
+  # platform (IRIX with clang).
+  CLIENT_LIBS += $(SDL_LIBS) -lm
+  RENDERER_LIBS = $(SDL_LIBS) -lm
+  ifeq ($(USE_CURL),1)
+    ifneq ($(USE_CURL_DLOPEN),1)
+      CLIENT_LIBS += $(CURL_LIBS)
+    endif
+  endif
+
 endif #Linux
 endif #darwin
 endif #MINGW
